halo2_proofs/src/circuit/floor_planner/v1/strategy.rs

```diff
@@ -194,12 +194,31 @@ fn slot_in(
     (regions, column_allocations)
 }
 
+// To preserve compatibility, when the "floor-planner-v1-legacy-pdqsort" feature is enabled,
+// we use a copy of the pdqsort implementation from the Rust 1.56.1 standard library, fixed
+// to its behaviour on 64-bit platforms.
+fn compatible_sort_by_key<K, F, T>(v: &mut Vec<T>, mut f: F)
+where
+    F: FnMut(&T) -> K,
+    K: Ord,
+{
+    // https://github.com/rust-lang/rust/blob/1.56.1/library/core/src/slice/mod.rs#L2365-L2402
+    #[cfg(feature = "floor-planner-v1-legacy-pdqsort")]
+    halo2_legacy_pdqsort::sort::quicksort(v, |a, b| f(a).lt(&f(b)));
+
+    #[cfg(not(feature = "floor-planner-v1-legacy-pdqsort"))]
+    v.sort_by_key(f);
```

Comment:
This is probably not what we want. It might be necessary to put the conditional code in `slot_in_biggest_advice_first` rather than factoring it out, even if that results in a bit uglier code.